cudax/include/cuda/experimental/__utility/memset_kernel.cuh

```diff
@@ -0,0 +1,59 @@
+
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__UTILITY_MEMSET_KERNEL_CUH
+#  define _CUDAX__UTILITY_MEMSET_KERNEL_CUH
+
+#  include <cuda/__cccl_config>
+
+#  if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#    pragma GCC system_header
+#  elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#    pragma clang system_header
+#  elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#    pragma system_header
+#  endif // no system header
+
+#  include <cuda/experimental/__launch/launch.cuh>
+#  include <cuda/experimental/__stream/stream_ref.cuh>
+
+#  include <cuda/std/__cccl/prologue.h>
+
+namespace cuda::experimental::__detail
+{
+template <typename T, typename Config>
+__global__ static void __memset_kernel(Config __config, T* __ptr, T __value, ::cuda::std::size_t __count)
```

Comment:
concern: Implementing our own kernel vs using Thrust doesn't actually solve the problem here. It just reintroduces the same problem that the Thrust arch namespace thing was introduced to solve. 

My understanding from recent discussions is that **no member function of an ABI-stable type should ever launch a kernel** and all functions that can launch kernels must be free functions